app/routes/admin.settings.tsx

```diff
@@ -312,9 +312,24 @@ export const action = async ({ request }: ActionFunctionArgs) => {
       } else if (body.get('intent') === 'delete-email-template') {
         await StoreConfig.deleteEmailTemplateByName(String(body.get('name')))
         return json({ error: null, data: {} }) //for modal dismissal
+      } else if (body.get('intent') === 'create-category') {
+        await new ProductCategoryModel().create({
+          id: '',
+          name: String(body.get('name')),
+          slug: String(body.get('slug')),
+        })
+      } else if (body.get('intent') === 'update-category') {
+        await new ProductCategoryModel().update({
+          id: payload.id,
```

Comment:
`payload.id` is the customer ID not the product category ID. I think you'll need to figure out a way to pass the product ID, such as ```<Input type="hidden" id="product-id" name="product-id" value={<the actual product ID>} />```

Reply:
done